src/DBFTables.jl

```diff
@@ -1,150 +1,293 @@
+    miss(tryparse(T, str))
+# String to avoid returning SubString{WeakRefString{UInt8}}
+function dbf_value(T::Type{String}, str::AbstractString)
+    stripped = rstrip(str)
+    if isempty(stripped)
+        # return missing rather than ""
+        return missing
+    else
+        return String(stripped)
+    end
+end
+dbf_value(T::Type{Nothing}, str::AbstractString) = missing
+
+# define get functions using getfield since we overload getproperty
+"Access the header of a DBF Table"
+getheader(dbf::Table) = getfield(dbf, :header)
+getfields(dbf::Table) = getheader(dbf).fields
+getstrings(dbf::Table) = getfield(dbf, :strings)
+getrow(row::Row) = getfield(row, :row)
+gettable(row::Row) = getfield(row, :table)
+
+Base.length(dbf::Table) = Int(getheader(dbf).records)
+Base.size(dbf::Table) = (length(dbf), length(getfields(dbf)))
+Base.size(dbf::Table, i) = size(dbf)[i]
+
+"""
+	DBFTables.Table(source) => DBFTables.Table
+
+Read a source, a path to a file or an opened stream, to a DBFTables.Table.
+This type conforms to the Tables interface, so it can be easily converted
+to other formats. It is possible to iterate through the rows of this object,
+or to retrieve columns like `dbf.fieldname`.
+"""
+function Table(io::IO)
+    header = Header(io)
+    # consider using mmap here for big dbf files
+    data = Vector{UInt8}(undef, header.rsize * header.records)
+    read!(io, data)
+    strings = _create_stringarray(header, data)
+    Table(header, data, strings)
+end
+
+function Table(path::AbstractString)
+    open(path) do io
+        Table(io)
+    end
+end
+
+"Collect all the offsets and lenghts from the header to create a StringArray"
+function _create_stringarray(header::Header, data::AbstractVector)
+    # first make the lengths and offsets for a single record
+    lengths_record = UInt32.(getfield.(header.fields, :length))
+    offsets_record = vcat(0, cumsum(lengths_record)[1:end-1]) .+ 1
+
+    # the lengths are equal for each record
+    lengths = repeat(lengths_record, 1, header.records)
+    # the offsets accumulate over records with the record size
+    row_offsets = range(0; length = header.records, step = header.rsize)
+    offsets = repeat(offsets_record, 1, header.records)
+    offsets .+= reshape(row_offsets, 1, :)
+
+    StringArray{WeakRefString{UInt8},2}(data, offsets, lengths)
+end
+
+"Create a NamedTuple representing a single row"
+function Base.NamedTuple(row::Row)
+    dbf = gettable(row)
+    str = getstrings(dbf)
+    fields = getfields(dbf)
+    ncol = length(fields)
+    rowidx = getrow(row)
+    @inbounds record = @view str[:, rowidx]
+    @inbounds prs = (fields[col].name => dbf_value(
+        fields[col].type,
+        record[col],
+    ) for col = 1:ncol)
+    return (; prs...)
+end
+
+function Base.show(io::IO, row::Row)
+    show(io, NamedTuple(row))
+end
+
+function Base.show(io::IO, dbf::Table)
+    nr = length(dbf)
+    nc = length(getfields(dbf))
+    println(io, "DBFTables.Table with $nr rows and $nc columns")
```

Comment:
Might be useful to show the `Tables.Schema` here? To show the column names/types?